### PR TITLE
feat: add blog header

### PR DESCRIPTION
**Pull request by georgiee**
Relates to #220 

---

This is part 1 of 2 refactoring the `HeaderBlock` into two distinct components. Here the Blog Header to be used exclusively in the blog as the header (hence the name).

I basically took the existing implementation, removed the obsolete parts and refactored the rest. I updated the naming following the guide pdf in the linked issues (kicker, byline, leader).

I skipped the `.stories.ts` file as we should use not both but one of the story formats which works pretty good.

I like how the component feels from the API usage and I was able to make the implementation more compact. Thanks @milljoniaer for the existing implementation, I was able to pick some styles easily 👍

## Changes
+ I focused on html tags. Something like that `const Headline = styled.p` is really bad from an accessible view and we should not let thigns like that slip into our codebase 🙏 In addition I based the entire component on a `header` tag plus `h1` to provide a correct markup in the context of blog posts. This also means this component is not meant to be used somewhere else.

+ I removed the entire exception handling of the `large` display. This is not relevant at all for the blog article and I will take a very careful look on this when working on part 2.

+ renamed the different parts (meta -> byLine, topline -> kicker and such) which feels much more solid now 💪
+ I moved some margins around so we don't end up with obsolete margins if an element is removed (e.g. Topline -> Headline, margin 16px is now on the topline)
+ I found a color with alpha 50%. That's not good we do not work with alpha values in text. I have normalized it to a hex value without alpha and commented in Figma.

Part 2 will cover the content-related component with headline + icon + text. I hope you see how divide and conquer helps providing two very useful components instead of one big blob of features that correlate in markup but not semantically (a blog header is never a content section).

### CODE REVIEW

File: src/new-components/blog-header/blog-header.tsx
Comment (hofmann-felix) [inline src/new-components/blog-header/blog-header.tsx:39]: Typography does not match the design. Applying the timestamp text style should solve it: 

```suggestion
const BylineStyled = styled.span`
  ${TextStyles.timestamp}
```

File: src/new-components/blog-header/blog-header.tsx
Comment (hofmann-felix) [inline src/new-components/blog-header/blog-header.tsx]: The design uses another font style on mobile: 

```suggestion
const LeadStyled = styled.div`
  ${TextStyles.textR}
  margin-top: 48px;

  ${up('md')} {
    margin-top: 32px;
    ${TextStyles.textL}
  }
`;
```


Edit: The headline is also smaller on mobile (Headline L instead of XL)

**Review by hofmann-felix — CHANGES_REQUESTED** [review]
Useful changes, especially regarding the accessibility and readability of the code 👍 In two places the font does not fit.

**Review by fischeversenker — APPROVED** [review]
I agree with the changes that @hofmann-felix requested. Apart from that I'm approving this.

File: src/new-components/blog-header/blog-header.tsx
Comment (georgiee) [reply src/new-components/blog-header/blog-header.tsx:39]: thanks for the catch and the helpful tip 🙏

**Comment by georgiee** [comment]
@hofmann-felix I did not notice at first but you made suggestion for the actual fixes. Thank you for that, this really helped moving forward while being busy with other things. I merged them and I'm happy to merge once approved.

**Review by hofmann-felix — APPROVED** [review]

**Pull request merged**
